gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/EntityServiceAndGatewayServiceConverter.java

```diff
@@ -165,19 +166,18 @@ private static Function convertToQueryFunction(
             .setFunctionName(function.getFunction().name())
             .setAlias(function.getAlias());
 
-    // AVGRATE is adding a specific implementation because Pinot does not directly support this
-    // function
     switch (function.getFunction()) {
       case AVGRATE:
         {
-          builder.setFunctionName(FunctionType.SUM.name()).setAlias(function.getAlias());
-
-          // Adds only the argument that is a column identifier for now.
-          List<org.hypertrace.gateway.service.v1.common.Expression> columns =
-              function.getArgumentsList().stream()
-                  .filter(org.hypertrace.gateway.service.v1.common.Expression::hasColumnIdentifier)
-                  .collect(Collectors.toList());
-          columns.forEach(e -> builder.addArguments(convertToEntityServiceExpression(e)));
+          builder.setFunctionName(function.getFunction().name()).setAlias(function.getAlias());
+          if (function.getArgumentsCount() > 0) {
```

Comment:
Just curious: Shouldn't there be a single argument if the validator has validated? If, that is the case, do we have to go over it all?

Reply:
For readability, as only literal would like to convert to iso duration, can we do something like below?
```
function.getArgumentsList().stream()
              .map(e -> e.hasLiteral() ? convertLongToIsoFormat(e) : e)
              .forEach(e -> builder.addArguments(convertToEntityServiceExpression(e)));
```